src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxrResolver.cpp

```diff
@@ -56,12 +59,46 @@ HostFxrResolver::GetHostFxrParameters(
             throw InvalidOperationException(L"Application arguments are empty.");
         }
 
+        BOOL gotHostFxrPath = FALSE;
         if (dotnetExePath.empty())
         {
-            dotnetExePath = GetAbsolutePathToDotnet(applicationPhysicalPath, expandedProcessPath);
+            // need to find dotnet for get_host_fxr_path when dotnet is launched from the path
```

Comment:
I'm not sure why we care about this scenario... would love to know if anyone uses it. We could get rid of a lot of complexity in our code if we don't have to worry about loading dotnet when it's not properly installed/registered.